packages/ui/src/views/repo/repo-summary/index.tsx

```diff
@@ -0,0 +1,221 @@
+        description?: string
+      }
+    | undefined
+  handleCreateToken: () => void
+  // TODO: fix this
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  repoEntryPathToFileTypeMap: Map<string, any>
+  // TODO: fix this
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  files: any[]
+  decodedReadmeContent: string
+  summaryDetails: {
+    default_branch_commit_count?: number
+    branch_count?: number
+    tag_count?: number
+    pull_req_summary?: { open_count: number } | undefined
+  }
+  spaceId: string
+  repoId: string
+  gitRef?: string
+  latestCommitInfo?: {
+    userName: string
+    message: string
+    timestamp: string
+    sha: string | null
+  }
+}
+
+export function RepoSummaryView({
+  loading,
+  selectedBranch,
+  branchList,
+  tagList = [],
+  selectBranch,
+  filesList,
+  navigateToFile,
+  repository,
+  handleCreateToken,
+  repoEntryPathToFileTypeMap,
+  files,
+  decodedReadmeContent,
+  summaryDetails: { default_branch_commit_count = 0, branch_count = 0, tag_count = 0, pull_req_summary },
+  spaceId,
+  repoId,
+  gitRef,
+  latestCommitInfo
+}: RepoSummaryViewProps) {
+  const navigate = useNavigate()
+
+  const renderListContent = () => {
+    if (loading) return <SkeletonList />
+
+    if (!loading && repoEntryPathToFileTypeMap.size > 0) {
+      return (
+        <Summary
+          latestFile={{
+            user: { name: latestCommitInfo?.userName || '' },
+            lastCommitMessage: latestCommitInfo?.message || '',
+            timestamp: latestCommitInfo?.timestamp || '',
+            sha: latestCommitInfo?.sha || ''
+          }}
+          files={files}
+        />
+      )
+    }
+
+    return (
+      <NoData
+        insideTabView
+        iconName="no-data-folder"
+        title="No files yet"
+        description={['There are no files in this repository yet.', 'Create new or import an existing file.']}
+        primaryButton={{ label: 'Create file' }}
+        secondaryButton={{ label: 'Import file' }}
+      />
+    )
+  }
+
+  return (
+    <SandboxLayout.Main hasLeftPanel hasHeader hasSubHeader>
+      <SandboxLayout.Columns columnWidths="1fr 220px">
+        <SandboxLayout.Column>
+          <SandboxLayout.Content>
+            <ListActions.Root>
+              <ListActions.Left>
+                <ButtonGroup.Root>
+                  <BranchSelector
+                    name={selectedBranch}
+                    branchList={branchList}
+                    tagList={tagList}
+                    selectBranch={selectBranch}
+                  />
+                  <SearchFiles navigateToFile={navigateToFile} filesList={filesList} />
+                </ButtonGroup.Root>
+              </ListActions.Left>
+              <ListActions.Right>
+                <ButtonGroup.Root>
+                  <DropdownMenu>
+                    <DropdownMenuTrigger asChild>
+                      <Button variant="outline">
+                        Add file&nbsp;&nbsp;
```

Comment:
Probably best to use a bit of left margin on the icon or introduce a flex layout with gap to the button, rather than use non-breaking spaces to push it out